core/input/input.h

```diff
@@ -100,30 +100,30 @@ class Input : public Object {
 	int64_t mouse_window = 0;
 	bool legacy_just_pressed_behavior = false;
 
-	struct Action {
+	struct ActionState {
 		uint64_t pressed_physics_frame = UINT64_MAX;
 		uint64_t pressed_process_frame = UINT64_MAX;
 		uint64_t released_physics_frame = UINT64_MAX;
 		uint64_t released_process_frame = UINT64_MAX;
-		uint64_t pressed = 0;
 		bool exact = true;
-		float strength = 0.0f;
-		float raw_strength = 0.0f;
-		LocalVector<float> strengths;
-		LocalVector<float> raw_strengths;
-
-		Action() {
-			strengths.resize(MAX_EVENT + 1);
-			raw_strengths.resize(MAX_EVENT + 1);
-
-			for (uint64_t i = 0; i <= MAX_EVENT; i++) {
-				strengths[i] = 0.0;
-				raw_strengths[i] = 0.0;
-			}
-		}
+
+		struct DeviceState {
+			bool pressed[MAX_EVENT + 1] = { false };
```

Comment:
```suggestion
			bool pressed[MAX_EVENT] = { false };
```
You can change MAX_EVENT to 32 or whatever number now, since it's used only here.